discord/primary_guild.py

```diff
@@ -0,0 +1,76 @@
+"""
+The MIT License (MIT)
+
+Copyright (c) 2021-present Pycord Development
+
+Permission is hereby granted, free of charge, to any person obtaining a
+copy of this software and associated documentation files (the "Software"),
+to deal in the Software without restriction, including without limitation
+the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the
+Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+"""
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from .state import ConnectionState
+
+from .asset import Asset
+from .types.primary_guild import PrimaryGuild as PrimaryGuildPayload
+
+
+class PrimaryGuild:
+    """
+    Represents a Discord Primary Guild.
+
+    .. versionadded:: 2.7
+
+    Attributes
+    ----------
+        identity_guild_id: int
+            The ID of the guild.
+        identity_enabled: :class:`bool`
+            Whether the primary guild is enabled.
+        tag: str
+            The tag of the primary guild.
+    """
+
+    def __init__(self, data: PrimaryGuildPayload, state: "ConnectionState") -> None:
+        self.identity_guild_id: int = data["identity_guild_id"]
+        self.identity_enabled: bool | None = data.get("identity_enabled", None)
+        self.tag: str = data["tag"]
+        self._badge: str = data["badge"]
+        self._state: "ConnectionState" = state
+
+    def __repr__(self) -> str:
+        return f"<PrimaryGuild identity_guild_id={self.identity_guild_id} identity_enabled={self.identity_enabled} tag={self.tag}>"
+
+    @property
+    def badge(self) -> Asset | None:
+        """Returns the badge asset, if available.
+
+        .. versionadded:: 2.7
+
+        .. note::
+            This information is only available via :meth:`Client.fetch_user`.
```

Comment:
that's fkin stupid  (thanks discord)

Reply:
eh copy paste error here. that data is always present

Reply:
actually i don't think this is true? works fine on my bot cache, i think this only applies to banner and accent color (not sure about nameplates/accessories)